hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/recovery/NMLeveldbStateStoreService.java

```diff
@@ -1813,4 +1813,22 @@ protected void checkVersion() throws IOException {
             + getCurrentVersion() + ", but loading version " + loadedVersion);
     }
   }
+  @Override
+  public void releaseAssignedResources(ContainerId containerId, String resourceType)  {
+    LOG.info("Release assigned resources for a container of a specified resourceType");
+    try {
+      try (WriteBatch batch = db.createWriteBatch()) {
+        String key = CONTAINERS_KEY_PREFIX + containerId
+            + CONTAINER_ASSIGNED_RESOURCES_KEY_SUFFIX + resourceType;
+        batch.delete(bytes(key));
+        db.write(batch);
+      }catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }catch (DBException e){
```

Comment:
Change this part similar to how storeAssignedResources handle the exception. LOG.error is not needed here as markStoreUnHealthy does the error logging.

 catch (DBException e) {
      markStoreUnHealthy(e);
      throw new IOException(e);
}

Reply:
made changes accordingly